src/read_write/codec.rs

```diff
@@ -137,3 +139,15 @@ where
 {
     num::cast::<T, f64>(value).unwrap() * edge_length + min
 }
+
+pub fn attribute_to_proto(attribute: &AttributeData) -> proto::AttributeDataType {
```

Comment:
Nit: This could be a method of AttributeDataType

Reply:
In src/lib.rs

Reply:
good idea, done!